content/components/button.mdx

```diff
@@ -0,0 +1,149 @@
+The default button is the default variant in the `Button` component.
+Here is what it looks like in different states.
+<Box>
+    <img width='400' src='https://user-images.githubusercontent.com/417268/141061136-44634e00-1105-465c-9f60-f8b2a0378ac3.png' />
+</Box>
+
+## Buttons with Icons
+
+Icons are used in Buttons(at Github) for:
+
+- A Leading Icon may be used to enhance the context of the button text
+- A Leading Icon may be used to denote a certain state of the button (eg Watch/Unwatch)
+- A Trailing Icon is not very common in usage, but the `TriangleDownIcon` icon is used in buttons as a base for the dropdown menu.
+
+Here's what a leading icon will look like in all of the variants
+
+<Box display="flex" alignItems="flex-start" flexDirection={["column", "column"]} sx={{gap: 2}}>
+    <span>Default</span>
+    <img width='200' src='https://user-images.githubusercontent.com/417268/141068226-e383a7fc-4845-4e29-b19a-cdedd7d1ba5f.png' />
+    <span>Primary</span>
+    <img width='200' src='https://user-images.githubusercontent.com/417268/141068584-4f8267df-1b5b-4a8d-9800-4651994b7664.png' />
+    <span>Invisible</span>
+    <img width='200' src='https://user-images.githubusercontent.com/417268/141068701-ce43bcc1-0e24-437d-810d-ae8afd928308.png' />
+    <span>Danger</span>
+    <img width='200' src='https://user-images.githubusercontent.com/417268/141068654-04a67134-568c-4ba9-9d88-85ba6c3a5bfa.png' />
+</Box>
+
+## Icon only Buttons
+
+These buttons will display only the icon component and visually hide the button text. They are used for popular actions like search/upvote or burger family menus.
+
+<Box>
+    <img width='300' src='https://user-images.githubusercontent.com/417268/141069418-ee6585f7-9941-410f-99ab-058494971b54.png' />
+</Box>
+
+## Button Counters
+
+Counters at the trailing end of the button denote extra context relevant to the button. Like upvote/watch count.
+We use the `CounterLabel` component to show this. 
+
+<Box display="flex" alignItems="flex-start" flexDirection={["column", "column"]} sx={{gap: 2}}>
+    <span>Default</span>
+    <img width='200' src='https://user-images.githubusercontent.com/417268/141070891-1781dc9e-2fab-4e73-bb08-29c9a8e4217f.png' />
+    <span>Primary</span>
+    <img width='200' src='https://user-images.githubusercontent.com/417268/141070946-7e7cfe29-9fcd-44aa-9bd0-c47ba27fa606.png' />
+    <span>Invisible</span>
+    <img width='200' src='https://user-images.githubusercontent.com/417268/141070963-f31aa823-dd2b-4eec-89bf-a9eab4a37a52.png' />
+    <span>Danger</span>
+    <img width='200' src='https://user-images.githubusercontent.com/417268/141070980-e34c3e8b-e765-49c8-807e-8e4a0eff8ddc.png' />
+</Box>
+
+
+## ButtonLink
+
+The ButtonLink will be a separate component. This will render as an anchor or an appropriate framework link component as configured by the user.
+
+// Waiting for design input from Ash
+
+## Accessible Buttons
+
+The Button component has to follow below criteria to be accessible
```

Comment:
@alliethu can you review this section? And also consider if this is the best place for this info to be.